src/integrationTest/java/uk/gov/hmcts/reform/pip/account/management/controllers/AccountTest.java

```diff
@@ -532,6 +562,41 @@ void testCreateMultipleSuccessUsers() throws Exception {
             objectMapper.readValue(response.getResponse().getContentAsString(),
                                    new TypeReference<>() {});
 
+        assertEquals(1, mappedResponse.get(CreationEnum.CREATED_ACCOUNTS).size(), "1 Users should be created");
+    }
+
+    @Test
+    void testCreateMultipleSuccessUsersWithDifferentEmails() throws Exception {
+
+        when(graphClient.users(any())).thenReturn(userRequestBuilder);
+        when(userRequestBuilder.buildRequest()).thenReturn(userRequest);
+        when(userRequest.get()).thenThrow(graphServiceException);
+
+        MockHttpServletRequestBuilder mockHttpServletRequestMediaUserBuilder = MockMvcRequestBuilders
+            .get(CREATE_MEDIA_USER_URL)
+            .contentType(MediaType.APPLICATION_JSON);
+
+        mockMvc.perform(mockHttpServletRequestMediaUserBuilder).andExpect(status().isOk()).andReturn();
+
+        PiUser validUser1 = createUser(true, UUID.randomUUID().toString());
+        PiUser validUser2 = new PiUser();
+        validUser2.setEmail("a@test.com");
+        validUser2.setProvenanceUserId(UUID.randomUUID().toString());
+        validUser2.setUserProvenance(PROVENANCE);
+        validUser2.setRoles(ROLE);
+
+        MockHttpServletRequestBuilder mockHttpServletRequestBuilder =
+            MockMvcRequestBuilders
+                .post(PI_URL)
+                .content(objectMapper.writeValueAsString(List.of(validUser1, validUser2)))
+                .header(ISSUER_HEADER, ISSUER_EMAIL)
+                .contentType(MediaType.APPLICATION_JSON);
+
+        MvcResult response = mockMvc.perform(mockHttpServletRequestBuilder).andExpect(status().isCreated()).andReturn();
+        ConcurrentHashMap<CreationEnum, List<Object>> mappedResponse =
+            objectMapper.readValue(response.getResponse().getContentAsString(),
+                                   new TypeReference<>() {});
+
```

Comment:
looking at account service code complexity i think it would be best to include int test that tests adding multiple users some valid and some duplicate

Reply:
testCreateMultipleSuccessUsers (This test is trying to create two users with same email address, but it creates only one because second is categorised as duplicate).
testCreateMultipleSuccessUsersWithDifferentEmails (this test creates two different users having different email address)

Reply:
i cant see the outcome you described in those tests, all the tests only test for 1 outcome, errored account,
![Screenshot 2022-07-04 at 09 38 04](https://user-images.githubusercontent.com/26488566/177116571-8f6f8097-1f4a-4b63-b747-9e9b621f91d5.png)
i can only see this outcome in the tests youve added, the created accounts is an empty list,